wgpu-hal/src/gles/adapter.rs

```diff
@@ -293,6 +321,22 @@ impl super::Adapter {
             ver >= (3, 1),
         );
 
+        // Check platforms effected by mesa bug:
+        // https://gitlab.freedesktop.org/mesa/mesa/-/issues/2565.
+        //
+        // This comment
+        // (https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/4972/diffs?diff_id=75888#22f5d1004713c9bbf857988c7efb81631ab88f99_323_327)
+        // seems to indicate all skylake models are effected, so we check for renderer models that
+        // have their 3 letter code ending with `L`
+        let r = renderer.to_lowercase();
+        if r.contains("mesa")
+            && r.contains("intel")
```

Comment:
is it possible to check for Vendor ID here instead of a string? or would WebGL not give us the vendor id?

Reply:
Unfortunately it seems to have no concept of a vendor ID :(